### ISSUE TITLE
What does 'Accuracy' mean in Regression?

### ISSUE THREAD

**Issue by Kevin-Moon** [keras-team/keras#7947]
Hello, Keras

I appreciate for this useful and great wrapper.

and I am building a network for the regression problem.

I saw that Keras calculate Acc and Loss even in regression.

how is it calculated?

loss moves alike MSE but the values are different to each other.

thank you

**Comment by WittmannF** [comment]
Is there any pull request for implementing R2 Score nativelly in keras? In order to have something like `model.compile(optimizer="Nadam", loss="mean_squared_error", metrics=["r2_score"])`?

**Comment by NichtJens** [comment]
The Keras docs about [custom metrics](https://keras.io/metrics/#custom-metrics) say (emphasis mine):

> The function would need to take (y_true, y_pred) as arguments and **return a single tensor value.**

The metrics/loss functions given here do not seem to do that, e.g.:

```
def rmse(y_true, y_pred):
    from keras import backend
    return backend.sqrt(backend.mean(backend.square(y_pred - y_true), axis=-1))
```

Here, `backend.mean()` is applied along the last axis only, thus all axes but the last axis will stay intact. I would think there should be a `backend.mean()` over all axes around the return value. Like so:

```
def rmse(y_true, y_pred):
    from keras import backend
    return backend.mean(
        backend.sqrt(backend.mean(backend.square(y_pred - y_true), axis=-1))
    )
```

Is that not needed?

Also, why is the backend imported inside each function. Wouldn't a single regular import at the top be sufficient as well?

**Comment by UnixJunkie** [comment]
@WittmannF I would even have the choice to use (1 - R2) as the loss and R2 as the performance metric.

**Comment by premkumar1303** [comment]
def r_square_loss(y_true, y_pred):
    from keras import backend as K
    SS_res =  K.sum(K.square(y_true - y_pred)) 
    SS_tot = K.sum(K.square(y_true - K.mean(y_true))) 
    return 1 - ( 1 - SS_res/(SS_tot + K.epsilon()))

Is this correct for data having multiple variables/attributes in time series forecasting?

**Comment by profPlum** [comment]
@fchollet It seems like no-one has answered the original question: in keras what is the 'accuracy' metric computing in regression problems?

I would still very much like to know, because it seems meaningful given that it is always non-zero (even for regression problems) & increases at a reasonable rate w.r.t. loss. But I do not know if I can trust it until I know what it is actually doing.

**Comment by Karric** [comment]
I have the same issue, my regression model is reporting a low 'accuracy' but it seems to perform fine on actual input.
